packages/embark-test-runner/src/index.js

```diff
@@ -235,9 +235,20 @@ class TestRunner {
                 if (global.embark.needConfig) {
                   global.config({});
                 }
-                global.embark.onReady((_err, accounts) => {
+
+                function runDescribe(accounts) {
                   self.ogMochaDescribe(describeName, callback.bind(mocha, accounts));
+                  setImmediate(() => {
+                    global.run(); // Call `run` in setImmediate to make sure tests with no `config()` don't hang
+                  });
                   global.run(); // This tells mocha that it can run the test (used in conjunction with `delay()`
```

Comment:
Nah, you're seeing right. It does run twice. I guess I forgot to add a comment as to why it's ok to call twice, but it boils down to: once Mocha is running, calling run again doesn't do anything, but in our case, it might be that the first run is too soon, so the second is fine, but sometimes the opposite.